spec/models/customer_spec.rb

```diff
@@ -1,7 +1,40 @@
 require 'rails_helper'
 
 RSpec.describe Customer do
+  before :each do
+    FactoryBot.reload
+    @customers = FactoryBot.create_list(:customer_with_invoice, 10)
+    @invoices = Invoice.all
+    FactoryBot.create_list(:transaction, 6, result: 'success', invoice_id: @invoices[0].id)
+    FactoryBot.create_list(:transaction, 5, result: 'success', invoice_id: @invoices[3].id)
+    FactoryBot.create_list(:transaction, 4, result: 'success', invoice_id: @invoices[1].id)
+    FactoryBot.create_list(:transaction, 3, result: 'success', invoice_id: @invoices[4].id)
+    FactoryBot.create_list(:transaction, 2, result: 'success', invoice_id: @invoices[2].id)
+  end
+
   describe "Relationships" do
     it {should have_many :invoices} 
+    it {should have_many(:transactions).through(:invoices)}
+  end
+
+  describe '#successful_transactions' do
```

Comment:
Would this follow SRP more closely if it was a transaction model method?

Reply:
Actually, this fulfills the user story well here. 👏🏽

Reply:
It definitely might!  We'll see what Brian says - I'll circle back to refactor if need be